src/components/TransactionHistoryRows/TransactionContract.tsx

```diff
@@ -35,18 +39,31 @@ export const TransactionContract = ({
     isReceive && !txDetails.tx.data?.method ? txDetails.tradeTx?.type : txDetails.tx.data?.method
   const isFirstAssetOutgoing = interactsWithWithdrawMethod && isSend
 
+  const isRevoke = useMemo(
+    () =>
+      Boolean(
+        txMetadata?.method === 'approve' && txMetadata.value && bn(txMetadata.value).isZero(),
+      ),
+    [txMetadata],
+  )
   const titlePrefix = translate(
-    txDetails.tx.data?.parser
-      ? `transactionRow.parser.${txDetails.tx.data?.parser}.${i18n}`
-      : 'transactionRow.unknown',
+    (() => {
+      if (txDetails.tx.data?.parser) {
+        return `transactionRow.parser.${txDetails.tx.data?.parser}.${isRevoke ? 'revoke' : i18n}`
+      }
+      return 'transactionRow.unknown'
+    })(),
   )
 
+  // TODO: translation
```

Comment:
Was this a TODO for this PR?

Reply:
That's correct, but I'm not going to add it within this PR for two reasons:
1. I'm not yet sure yet which place this translation should live in (not a big deal, it can be in a "wrong" place there)
2. Different languages can have different syntax, and what is a suffix in English might be a prefix in e.g French or Spanish

The others (revoke/infinite) made sense to add as a translation, but this one doesn't as thinking linguistical implications is off the scope of this PR IMO!